.github/workflows/test.yml

```diff
@@ -0,0 +1,200 @@
+name: Test Suite
+
+on:
+  push:
+    branches: [ main, develop ]
+  pull_request:
+    branches: [ main, develop ]
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.25'
+    
+    - name: Cache Go modules
+      uses: actions/cache@v3
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    
+    - name: Run unit tests
+      run: task test-unit
+    
+    - name: Generate coverage report
+      run: |
+        go test -coverprofile=coverage.out ./utils/... ./horse/... ./user/... ./farm/...
+        go tool cover -html=coverage.out -o coverage.html
+    
+    - name: Upload coverage to artifacts
+      uses: actions/upload-artifact@v3
```

Comment:
@cursoragent this job fails to start because the version of the upload-artifact action is too far out of date. Please update it to the latest version, and ensure all the other actions are using the latest versions as well